app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,53 @@
+class ArticlesController < ApplicationController
+  
+  http_basic_authenticate_with name: "user", password: "secret", except: [:index, :show]
+
+  def index
+    @articles = Article.all
+  end
+
+  def show
+    @article = Article.find(params[:id])
+  end
+
+  def new
```

Comment:
What is the difference between `new` and `create` actions? Why do you need them both?

Reply:
With `new` , rails return a empty object to be created. And `create` check the necessary values to be stored in the object.

Reply:
What else? There are more differences.

Reply:
Another difference is that `create` with the `.save` stores the values in database, and if an error ocurrs render again the form with their respective errors.